snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedConceptSearchRequest.java

```diff
@@ -215,9 +222,12 @@ protected Expression prepareQuery(BranchContext context) {
 			
 			if (termFilter != null) {
 				try {
-					final ComponentCategory category = SnomedIdentifiers.getComponentCategory(termFilter.getTerm());
-					if (category == ComponentCategory.CONCEPT) {
-						conceptScoreMap.put(termFilter.getTerm(), Float.MAX_VALUE);
+					final String singleTerm = termFilter.getSingleTermOrNull();
```

Comment:
Don't we need scores for >1 terms as well? `getSingleTermOrNull` returns the term if there was exactly one given.

Reply:
Yeah, that is not supported yet here, not sure if we need it, this is backward compatible with the API functionality. I can add a note here for future selves.

Reply:
It is already handled by returning null in case of zero or more than 1 terms specified. I'll add a note though that multi-term ID matching is not supported and `filterById` should be used instead.